lib/OpenQA/CLI/api.pm

```diff
@@ -26,16 +26,30 @@ sub command ($self, @args) {
       'd|data=s' => \$data,
       'f|form' => \my $form,
       'j|json' => \my $json,
+      'H|hook-scripts=s' => \my $hooksrc,
       'param-file=s' => \my @param_file,
       'r|retries=i' => \my $retries,
       'X|method=s' => \(my $method = 'GET');
 
     @args = $self->decode_args(@args);
     die $self->usage unless my $path = shift @args;
 
-    $data = path($data_file)->slurp if $data_file;
-    my @data = ($data);
-    my $params = $form ? decode_json($data) : $self->parse_params(\@args, \@param_file);
+    my $params;
+    my @data;
+    if ($hooksrc) {
+	# if $hooksrc is a json  
+        my $hookdata = path("hook/$hooksrc.json")->slurp;
+	$data = ($hookdata);
```

Comment:
Please avoid needless copies. The use of parenthesis here makes no sense as well.